lib/PGObject/Util/DBAdmin.pm

```diff
@@ -559,26 +559,62 @@ sub server_version {
 }
 
 
-=head2 list_dbs([$dbname])
+=head2 list_dbs([$dbname],[$excludes])
 
 Returns a list of db names.
 
 When a database name is specified, uses that database to connect to,
 using the credentials specified in the instance.
 
 If no database name is specified, 'template1' is used.
+If an array of excluded databases is provided, it is used to limit the list
 
 =cut
 
 sub list_dbs {
     my $self = shift;
-    my $dbname = (shift @_) || 'template1';
+    my $dbname = shift || 'template1';
+    my $excludes = shift;
+
+    my $_query = 'SELECT datname from pg_database ' .
+                'WHERE datallowconn ' .
+                ( $excludes ? "AND datname NOT IN ('" . $excludes.join("','") . "') " : '' ) .
```

Comment:
Here we have Bobby Tables coming. I don't think we should do SQL by string composition. Does DBI/DBD::Pg support `IN (?)` where the argument is the array of values in which the values sit. if `IN (?)` doesn't work then maybe `IN (UNNEST(?))` does it?

Reply:
BTW, Perl doesn't "do" `$excludes.join()` (been doing Python lately? :-) )